File: src/main/java/at/hrechny/predictionsbot/connector/apifootball/exception/ApiFootballConnectorException.java
package at.hrechny.predictionsbot.connector.apifootball.exception;

public class ApiFootballConnectorException extends RuntimeException {

  public enum Reason {
    TOO_OFTEN_REQUESTS,
    QUOTA_EXCEEDED,
    REQUEST_ERROR,
    INVALID_RESPONSE
  }

  public ApiFootballConnectorException(Reason reason) {
    super(reason.toString());
  }

}

File: src/main/kotlin/at/hrechny/predictionsbot/connector/apifootball/exception/ApiFootballConnectorException.kt
package at.hrechny.predictionsbot.connector.apifootball.exception

class ApiFootballConnectorException(
    val reason: Reason,
    details: String? = null,
    cause: Throwable? = null,
) : RuntimeException(listOfNotNull(reason.toString(), details).joinToString(": "), cause) {
    enum class Reason {
        TOO_OFTEN_REQUESTS,
        QUOTA_EXCEEDED,
        REQUEST_ERROR,
        INVALID_RESPONSE,
    }
}
